### PR TITLE
Added solution to Project Euler problem 301

### PR DESCRIPTION
**Pull request by KumarUniverse**
### **Describe your change:**

* [x] Add an algorithm?
* [ ] Fix a bug or typo in an existing algorithm?
* [ ] Documentation change?

### **Checklist:**
* [x] I have read [CONTRIBUTING.md](https://github.com/TheAlgorithms/Python/blob/master/CONTRIBUTING.md).
* [x] This pull request is all my own work -- I have not plagiarized.
* [x] I know that pull requests will not be merged if they fail the automated tests.
* [x] This PR only changes one algorithm file.  To ease review, please open separate PRs for separate algorithms.
* [x] All new Python files are placed inside an existing directory.
* [x] All filenames are in all lowercase characters with no spaces or dashes.
* [x] All functions and variable names follow Python naming conventions.
* [x] All function parameters and return values are annotated with Python [type hints](https://docs.python.org/3/library/typing.html).
* [x] All functions have [doctests](https://docs.python.org/3/library/doctest.html) that pass the automated testing.
* [x] All new algorithms have a URL in its comments that points to Wikipedia or other similar explanation.
* [x] If this pull request resolves one or more open issues then the commit message contains `Fixes: #{$ISSUE_NO}`.

### CODE REVIEW

File: project_euler/problem_301/sol1.py [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+
+def solution(n: int = 2 ** 10) -> int:
+    """
+    For a given integer n <= 2^30, returns how many Nim games are lost.
+    >>> solution(2)
+    2
+    >>> solution(2 ** 10)
+    144
+    """
+    lossCount = 0
```
Comment (mateuszz0000) [inline project_euler/problem_301/sol1.py:66]: loss_count

**Review by mateuszz0000 — APPROVED** [review]

**Review by mateuszz0000 — DISMISSED** [review]
Missclick

**Comment by KumarUniverse** [comment]
Why does the  pre-commit run keep failing? The log say isort failed but I don't know what that means.

**Review by KumarUniverse — COMMENTED** [review]
Hi, I renamed my lossCount variable to loss_count. Can you review it please?

**Comment by KumarUniverse** [comment]
I managed to pass all 3 checks and made the requested changes. Can you please accept my pull request? Thank you.

File: project_euler/problem_301/sol1.py [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+    """
+    For a given integer n <= 2^30, returns how many Nim games are lost.
+    >>> solution(2)
+    2
+    >>> solution(2 ** 10)
+    144
+    """
+    loss_count = 0
+    for i in range(1, n + 1):
+        if x(i, 2 * i, 3 * i) == 0:
```
Comment (dhruvmanila) [inline project_euler/problem_301/sol1.py:68]: I don't think you need a function for such a small thing :)

File: project_euler/problem_301/sol1.py [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+            loss_count += 1
+
+    return loss_count
+
+
+if __name__ == "__main__":
+    print(solution())
```
Comment (dhruvmanila) [inline project_euler/problem_301/sol1.py:75]:
```suggestion
    print(f"{solution() = }")
```

**Comment by dhruvmanila** [comment]
Your solution is taking 3 minutes to run. You need to optimize it to bring it down to seconds. Try not to use brute force in here as time complexity matters in here.

https://travis-ci.com/github/TheAlgorithms/Python/jobs/400702745#L243

```
============================= slowest 10 durations =============================
167.57s call     project_euler/validate_solutions.py::test_project_euler[problem_301/sol1.py]
25.52s call     project_euler/validate_solutions.py::test_project_euler[problem_014/sol1.py]
20.23s call     project_euler/validate_solutions.py::test_project_euler[problem_014/sol2.py]
15.14s call     project_euler/validate_solutions.py::test_project_euler[problem_009/sol1.py]
11.44s call     project_euler/validate_solutions.py::test_project_euler[problem_010/sol1.py]
7.69s call     project_euler/validate_solutions.py::test_project_euler[problem_010/sol2.py]
5.45s call     project_euler/validate_solutions.py::test_project_euler[problem_034/sol1.py]
3.68s call     project_euler/validate_solutions.py::test_project_euler[problem_112/sol1.py]
3.67s call     project_euler/validate_solutions.py::test_project_euler[problem_012/sol1.py]
3.66s call     project_euler/validate_solutions.py::test_project_euler[problem_012/sol2.py]
```

**Comment by KumarUniverse** [comment]
I have optimized my solution and committed the changes.

File: project_euler/problem_301/sol1.py [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+    This function returns how many Nim games are lost given that
+    each Nim game has three heaps of the form (n, 2*n, 3*n).
+    >>> solution(0)
+    1
+    >>> solution(2)
+    3
+    >>> solution(10)
+    144
+    >>> solution(30)
+    2178309
```
Comment (dhruvmanila) [inline project_euler/problem_301/sol1.py:48]: You do not need to test the answer. It is being tested by our script. Please remove these two lines.

File: project_euler/problem_301/sol1.py [hunk 5]
```diff
@@ -0,0 +1,61 @@
...
+    """
+    # To find how many total games were lost for a given exponent x,
+    # we need to find the Fibonacci number F(x+2).
+    fib_ind = exponent + 2
```
Comment (dhruvmanila) [inline project_euler/problem_301/sol1.py:52]: Descriptive variable names please.

File: project_euler/problem_301/sol1.py [hunk 6]
```diff
@@ -0,0 +1,61 @@
...
+    phi = (1 + 5 ** 0.5) / 2
+    fib = (phi ** fib_ind - (phi - 1) ** fib_ind) / 5 ** 0.5
+    nim_loss_count = int(fib)
+
+    return nim_loss_count
```
Comment (dhruvmanila) [inline project_euler/problem_301/sol1.py:57]:
```suggestion

    return int(fib)
```

**Review by dhruvmanila — CHANGES_REQUESTED** [review]

**Comment by KumarUniverse** [comment]
I have made the requested changes to the code @dhruvmanila.

**Review by dhruvmanila — APPROVED** [review]
Thank you for your contribution!

**Pull request merged**